src/megaclient.cpp

```diff
@@ -3983,12 +3980,15 @@ bool MegaClient::procsc()
                             memset(&(it->second->changed), 0, sizeof it->second->changed);
                         }
 
-                        // historic user alerts are not supported for public folders
-                        if (sid.size())
+                        if (!loggedinfolderlink())
                         {
+                            // historic user alerts are not supported for public folders
                             // now that we have loaded cached state, and caught up actionpackets since that state
                             // (or just fetched everything if there was no cache), our next sc request can be for useralerts
                             useralerts.begincatchup = true;
+
+                            // now that we know all contacts, fetch public keys and check authentication
+                            fetchContactsKeys();
```

Comment:
I'm not sure about fetching the keys of contacts at this point - we have just decided we are `statecurrent`, so the app may expect everything including keys to be present at this point.   Also, do we need to fetch all keys of all contacts on every startup?  Hopefully these can be stored in cache, and we can just fetch those that need to be updated?   thanks

Reply:
Regarding this:
> do we need to fetch all keys of all contacts on every startup? Hopefully these can be stored in cache, and we can just fetch those that need to be updated? 
It is exactly the current behavior. The `getua()`/`getpubk()` looks for the attribute in the cache. If not found, it fetches from server.

As for fetching the keys at this point on every startup, I also doubt if it's really worthy. Apart from the first update for native users with missing authrings, where checking the authrings on resume session is desired, there could be some edge cases where a new contact or a key/signature has not been checked and the app is closed. In that case, keys won't be authenticated until a new session is created.
Taking into account the computation time must be very low for the authentication, I would prefer to keep it there on every startup (as Webclient does, according to my testing). But if you think it's too heavy, we can move it a bit upwards in the code and do authentication only upon new session.

Reply:
No, that's fine, thanks - I had not realised the getua() delivered from cache if it's present, rather than fetching from the servers every time :-)  cheers